rfc/0008-template.rst

```diff
@@ -0,0 +1,204 @@
+      values must be the reference allele first, then the alternate alleles as listed in the ALT
+      column.
+
+    - G: The field has one value for each possible genotype. The values must be in the same order as
+      prescribed in section 1.6.2 (see Genotype Ordering).
+
+    - . (dot): The number of possible values varies, is unknown or unbounded.
+
+Let's analyze the cases:
+
++------------------------------------+-----------------------------------------+
+|Number                              |Meaning of "."                           |
++====================================+=========================================+
+|0 or >=2                            |N/A                                      |
++------------------------------------+-----------------------------------------+
+|1                                   |N/A, because Number=1 is not an array.   |
++------------------------------------+-----------------------------------------+
+|A, with 0 or >=2 alternate alleles  |N/A                                      |
++------------------------------------+-----------------------------------------+
+|A with 1 alternate allele           |Ambiguous.                               |
++------------------------------------+-----------------------------------------+
+|R with >=1 alternate alleles        |N/A                                      |
++------------------------------------+-----------------------------------------+
+|R with 0 alternate alleles          |Ambiguous.                               |
++------------------------------------+-----------------------------------------+
+|G with >=1 alternate alleles        |N/A                                      |
++------------------------------------+-----------------------------------------+
+|G with 0 alternate alleles          |Ambiguous.                               |
++------------------------------------+-----------------------------------------+
+|.                                   |Ambiguous.                               |
++------------------------------------+-----------------------------------------+
+
+The R and G cases which are ambiguous can only occur at reference-only sites (because there are zero
+alternate alleles). This seems unlikely to happen in the wild as GVCFs typically have at least the
+``<NON_REF>`` alternate allele.
+
+The A case might be fairly common. We'll need a way for the user to specify how to interpet that.
+
+The "Number=." case is probably the most common ambiguous case. Again, the user will need to specify
+how to interpret that.
+
+We propose these modest modifcations to the `import_vcf` signature:
+
+1. Mark ``array_elements_required`` as deprecated.
+
+2. Add ``disambiguate_single_dot: Dict[str, Callable[[Expression], Expression]]`` which, for each
+   INFO field f, accepts a row at which field f was "." and returns a value for field f. The value
+   of other fields with a "." in this row is unspecified.
```

Comment:
Can you flesh this out a bit? I didn't understand on my first reading. Offline, we also discussed specifying the argument to the Callable to be a struct containing the entire current row, but with all fields in `disambiguate_single_dot` missing.

Reply:
@patrick-schultz addressed